### PR TITLE
Allow providing custom certificate and private key to https server

### PR DESCRIPTION
**Pull request by will-in-wi**
Adds feature requested in #197.

Thanks!

### CODE REVIEW

**Comment by gnufied** [comment]
Awesome. if you can make travis happy, I will merge!

File: lib/invoker/cli.rb [hunk 1]
```diff
@@ -41,11 +41,19 @@ def uninstall
       type: :boolean,
       banner: "Disable color in output",
       aliases: [:nc]
+    option :certificate,
+      type: :string,
+      banner: "Provide certificate for https (can include chain)"
```
Comment (gnufied) [inline lib/invoker/cli.rb]: lets rename banner to `path to certificate`

File: lib/invoker/cli.rb [hunk 2]
```diff
@@ -41,11 +41,19 @@ def uninstall
...
+    option :private_key,
+      type: :string,
+      banner: "Provide private key for https"
```
Comment (gnufied) [inline lib/invoker/cli.rb]: ditto

**Comment by will-in-wi** [comment]
The travis issue is a problem with Bundler 1.8.4. I can reproduce that locally, and it goes away if I update bundler.

The simplest fix would be to have the travis job upgrade bundler before running. Any issues with that?

**Comment by gnufied** [comment]
@will-in-wi yep, that is fine. merging

**Comment by rupurt** [comment]
You're a legend @will-in-wi!

@gnufied any plans for a new release with this awesome improvement?

**Comment by gnufied** [comment]
@rupurt new version is up please test and let us know.

**Comment by rupurt** [comment]
@gnufied I haven't been able to get it to work but I'm not sure if I'm generating the certificate correctly.

I've created the self signed cert with the instructions on [Heroku](https://devcenter.heroku.com/articles/ssl-certificate-self) using the common name `web.dev`. I've also imported it into the OS X System Keychain and set it to `Always Trust`.

Do you or @will-in-wi have any suggested instructions?

**Comment by rupurt** [comment]
I actually just tried it on Safari and it actually works! 😄 

It seems like Chrome + Firefox have aggressive caching and I haven't been able to get it to work in either even after doing a factory reset of the settings.

**Comment by will-in-wi** [comment]
Firefox has its own keychain. I'd expect Chrome to work, unless you are using a nightly or something. Chrome recently added `.dev` (since they technically own it) to their preloaded HSTS list. https://ma.ttias.be/chrome-force-dev-domains-https-via-preloaded-hsts/

**Comment by rupurt** [comment]
@will-in-wi thanks for the heads up. I managed to get it working across all browsers.

- The instructions from Heroku generate a x509 v1 certificate but v3 is required in Chrome 58+. I followed these [instructions](https://alexanderzeitler.com/articles/Fixing-Chrome-missing_subjectAltName-selfsigned-cert-openssl/) to generate a new v3 certificate with a `Subject Alternative Name`
- I got Firefox to work by importing the v3 certificate from above into the `Authorities` section of the `Certificate Manager`

**Pull request merged**